File: check_dashboard.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  const consoleMessages = [];
  const consoleErrors = [];
  
  page.on('console', msg => {
    const data = { type: msg.type(), text: msg.text() };
    consoleMessages.push(data);
    if (msg.type() === 'error') {
      consoleErrors.push(msg.text());
    }
  });
  
  try {
    await page.goto('http://localhost:8080', { timeout: 10000 });
    await page.waitForLoadState('networkidle', { timeout: 10000 });
    
    await page.screenshot({ path: '/tmp/dashboard_screenshot.png', fullPage: true });
    
    const domContent = await page.content();
    
    console.log('=== PAGE RENDERING ===');
    console.log('Page title:', await page.title());
    console.log('URL:', page.url);
    
    console.log('\n=== CONSOLE ERRORS ===');
    if (consoleErrors.length > 0) {
      consoleErrors.forEach(err => console.log('ERROR:', err));
    } else {
      console.log('No console errors');
    }
    
    console.log('\n=== CONSOLE MESSAGES (all) ===');
    consoleMessages.forEach(msg => console.log(`[${msg.type}] ${msg.text}`));
    
    console.log('\n=== DOM STRUCTURE (first 3000 chars) ===');
    console.log(domContent.substring(0, 3000));
    
    console.log('\n=== SCREENSHOT SAVED ===');
    console.log('Path: /tmp/dashboard_screenshot.png');
    
  } catch (e) {
    console.log('Error:', e.message);
  }
  
  await browser.close();
})();

File: check_dashboard.py
from playwright.sync_api import sync_playwright
import json


def check_dashboard():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        console_messages = []
        console_errors = []

        def handle_console(msg):
            console_messages.append({"type": msg.type, "text": msg.text})
            if msg.type == "error":
                console_errors.append(msg.text)

        page.on("console", handle_console)

        try:
            page.goto("http://localhost:8080", timeout=10000)
            page.wait_for_load_state("networkidle", timeout=10000)

            screenshot_path = "/tmp/dashboard_screenshot.png"
            page.screenshot(path=screenshot_path, full_page=True)

            dom_content = page.content()

            print("=== PAGE RENDERING ===")
            print(f"Page title: {page.title()}")
            print(f"URL: {page.url}")

            print("\n=== CONSOLE ERRORS ===")
            if console_errors:
                for err in console_errors:
                    print(f"ERROR: {err}")
            else:
                print("No console errors")

            print("\n=== CONSOLE MESSAGES (all) ===")
            for msg in console_messages:
                print(f"[{msg['type']}] {msg['text']}")

            print("\n=== DOM STRUCTURE (first 3000 chars) ===")
            print(dom_content[:3000])

            print(f"\n=== SCREENSHOT SAVED ===")
            print(f"Path: {screenshot_path}")

        except Exception as e:
            print(f"Error: {e}")

        browser.close()


if __name__ == "__main__":
    check_dashboard()
